src/viperleed/calc/files/parameters/file_reader.py

```diff
@@ -0,0 +1,61 @@
+"""Module file_reader of viperleed.calc.classes.
+
+This module is a more abstract version of a settings file reader. It is based on
+@michele-riva's work on the PARAMETERS file reader, which in turn is based on
+the old readPARAMETERS from @fkraushofer.
+
+Defines context-manager, iterator classes for reading parameters from a file.
+"""
+
+__authors__ = (
+    'Alexander M. Imre (@amimre)',
+    'Michele Riva (@michele-riva)',
+    )
+__copyright__ = 'Copyright (c) 2019-2024 ViPErLEED developers'
+__created__ = '2024-10-03'
+__license__ = 'GPLv3+'
+
+from abc import abstractmethod
+from collections.abc import Iterator
+from contextlib import AbstractContextManager
+from pathlib import Path
+
+
+class SettingsFileReader(AbstractContextManager, Iterator):
+    """A context manager that iterates the contents of a settings file
+    (e.g PARAMETERS)."""
+
+    def __init__(self, filename, noisy=True):                                   # TODO: it would be nice to support passing file contents via a StringIO or similar
```

Comment:
Generalization to a stream is simple. We could have a parent class that takes care of the stream (e.g., `InputStreamReader`). This one would derive from it. You could take a look at how we do it in `poscar`.